File: Shared/Preferences Controllers/MiscPreferenceController.m
//
//  MiscPreferenceController.m
//  PPMacho
//
//  Created by C.W. Betts on 7/26/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "MiscPreferenceController.h"
#import "UserDefaultKeys.h"

@implementation MiscPreferenceController
@synthesize addExt;
@synthesize madCompression;
@synthesize noLoadMixerFromFile;
@synthesize oscDrawLines;

+ (instancetype)newPreferenceView
{
	return [[self alloc] init];
}

- (IBAction)changePrefs:(id)sender
{
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];

	[defaults setBool:[addExt state] forKey:PPMAddExtension];
	[defaults setBool:[madCompression state] forKey:PPMMadCompression];
	[defaults setBool:[noLoadMixerFromFile state] forKey:PPMNoLoadMixerFromFiles];
	[defaults setBool:[oscDrawLines state] forKey:PPMOscilloscopeDrawLines];
	
	[[NSNotificationCenter defaultCenter] postNotificationName:PPMiscPreferencesDidChange object:self];
}

- (instancetype)init
{
	if (self = [super initWithNibName:@"MiscPrefs" bundle:nil]) {
		[self setTitle:NSLocalizedStringFromTable(@"Misc.", @"PreferenceNames", @"Misc.")];
	}
	return self;
}

- (void)awakeFromNib
{
	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
	[addExt setState:[defaults boolForKey:PPMAddExtension]];
	[madCompression setState:[defaults boolForKey:PPMMadCompression]];
	[noLoadMixerFromFile setState:[defaults boolForKey:PPMNoLoadMixerFromFiles]];
	[oscDrawLines setState: [defaults boolForKey:PPMOscilloscopeDrawLines]];
}

@end

File: Shared/Preferences Controllers/MiscPreferenceController.swift
//
//  MiscPreferenceController.swift
//  PPMacho
//
//  Created by C.W. Betts on 1/5/15.
//
//

import Cocoa
import SwiftAdditions
import MASPreferences.MASPreferencesViewController

final class MiscPreferenceController: NSViewController, MASPreferencesViewController {
	@IBOutlet weak var addExt: NSButtonCell!
	@IBOutlet weak var madCompression: NSButtonCell!
	@IBOutlet weak var noLoadMixerFromFile: NSButtonCell!
	@IBOutlet weak var oscDrawLines: NSButton!
	
	class func newPreferenceView() -> MiscPreferenceController? {
		let aCont = MiscPreferenceController(nibName: NSNib.Name("MiscPrefs"), bundle: nil)
		aCont.title = NSLocalizedString("Misc.", tableName: "PreferenceNames", comment: "Misc.")
		return aCont
	}
	
	override var identifier: NSUserInterfaceItemIdentifier? {
		get {
			return .PPMiscPrefID
		}
		set {
			//do nothing
		}
	}
	
	var toolbarItemLabel: String? {
		return NSLocalizedString("Misc.", tableName: "PreferenceNames", comment: "Misc.")
	}
	
	var toolbarItemImage: NSImage? {
		return NSImage(named: NSImage.Name.advanced)
	}

	override func awakeFromNib() {
		super.awakeFromNib()
		let defaults = UserDefaults.standard
		addExt.state = defaults.bool(forKey: PPMAddExtension) ? .on : .off
		madCompression.state = defaults.bool(forKey: PPMMadCompression) ? .on : .off
		noLoadMixerFromFile.state = defaults.bool(forKey: PPMNoLoadMixerFromFiles) ? .on : .off
		oscDrawLines.state = defaults.bool(forKey: PPMOscilloscopeDrawLines) ? .on : .off
	}
	
	@IBAction func changePrefs(_ sender: AnyObject?) {
		let defaults = UserDefaults.standard
		
		defaults[PPMAddExtension] = addExt.state == .on
		defaults[PPMMadCompression] = madCompression.state == .on
		defaults[PPMNoLoadMixerFromFiles] = noLoadMixerFromFile.state == .on
		defaults[PPMOscilloscopeDrawLines] = oscDrawLines.state == .on
		
		NotificationCenter.default.post(name: .PPMiscPreferencesDidChange, object: self)
	}
	
	var hasResizableWidth: Bool {
		return false
	}
	
	var hasResizableHeight: Bool {
		return false
	}
}
